src/Init/System/IO.lean

```diff
@@ -496,6 +496,29 @@ instance : MonadLift (ST IO.RealWorld) (EIO ε) := ⟨fun x s =>
 def mkRef (a : α) : IO (IO.Ref α) :=
   ST.mkRef a
 
+/-- Keeps an object alive by ensuring that its RC >= 1. Unlike holding typed `Ref`s,
+this structure is homogeneous. The ref can safely be shared between threads. -/
```

Comment:
> The ref can safely be shared between threads

Is this the only reason it's a ref at all? It doesn't look like we ever write to it.

Reply:
I think my reasoning in one of the outsourced-GC prototypes was that the *action* of incing the RC should really be an IO action so as to not get erased by any const compiler magic. But since we have an `aliveRefs` table now which keeps things alive, I think said table could just contain `NonScalar`s. Is this what you're suggesting?

Reply:
Sounds good